keps/sig-multicluster/1645-multi-cluster-services-api/README.md

```diff
@@ -450,154 +473,87 @@ about permitting exports from `kube-system` and `default`. As a best practice,
-  sync as expected. For example If you export a `NodePort` service, the
-  resulting cross-cluster service will still be a supercluster IP type. You
-  could use node ports to access the cluster-local service in the source
-  cluster, but not in any other cluster, and it would only route to local
-  endpoints.
-- `ExternalName`: It doesn't make sense to export an `ExternalName` service.
-  They can't be merged with other exports, and it seems like it would only
-  complicate deployments by even attempting to stretch them across clusters.
-  Instead, regular `ExternalName` type `Services` should be created in each
-  cluster individually.
-
-### Consumption of EndpointSlice
+### Importing Services
 
 To consume a supercluster service, users will use the domain name associated
-with their `ServiceExport`. When the mcsd-controller sees a `ServiceExport`, a
-`ServiceImport` will be introduced, which can be largely ignored by the user. 
-
-An `ServiceImport` is a service that may have endpoints in other clusters.
+with their `ServiceExport`. When the mcs-controller sees a `ServiceExport`, a
+`ServiceImport` will be introduced in each cluster to represent the imported
+service. Users are primarily expected to consume the service via domain name and
+supercluster VIP, but the `ServiceImport` may be used for imported service
+discovery via the K8s API and will be used internally as the source of routing
+and DNS configuration.
+
+A `ServiceImport` is a service that may have endpoints in other clusters.
 This includes 3 scenarios:
 1. This service is running entirely in different cluster(s)
 1. This service has endpoints in other cluster(s) and in this cluster
 1. This service is running entirely in this cluster, but is exported to other cluster(s) as well
 
 For each exported service, one `ServiceExport` will exist in each cluster that
-runs the service. The mcsd-controller will create and maintain a derived
+runs the service. The mcs-controller will create and maintain a derived
 `ServiceImport` in each cluster within the supercluster (see: [constraints and
-conflict resolution](#constraints-and-conflict-resolution)). If all `ServiceExport` instances are deleted, each
-`ServiceImport` will also be deleted from all clusters.
+conflict resolution](#constraints-and-conflict-resolution)). If all
+`ServiceExport` instances are deleted, each `ServiceImport` will also be deleted
+from all clusters.
 
 Since a given `ServiceImport` may be backed by multiple `EndpointSlices`, a
 given `EndpointSlice` will reference its `ServiceImport` using the label
 `multicluster.kubernetes.io/imported-service-name` similarly to how an
 `EndpointSlice` is associated with its `Service` in a single cluster. Each
 imported `EndpointSlice` will also have a
-`multicluster.kubernetes.io/source-cluster` label with a registry-scoped unique
-identifier for the cluster.
+`multicluster.kubernetes.io/source-cluster` label with the cluster name, a
```

Comment:
They do not necessarily correspond 1:1 (but may). Each EndpointSlice maps only to a single source cluster, but it's possible that two slices from the same source cluster are combined. This could mean N*M endpoint slices exist.

Imported slice lifecycle is managed by the mcs-controller, so it's up to the implementation to avoid naming conflicts. I don't think we need to dictate a specific format